skrub/_matching.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+from scipy import sparse
+from sklearn.base import BaseEstimator
+from sklearn.neighbors import NearestNeighbors
+
+
+class Matching(BaseEstimator):
+    def __init__(self):
+        pass
+
+    def fit(self, aux, main):
+        del main
+        self.aux_ = aux
+        self.neighbors_ = NearestNeighbors(n_neighbors=1).fit(aux)
+        return self
+
+    def match(self, main, max_dist):
+        distances, indices = self.neighbors_.kneighbors(main, return_distance=True)
+        distances, indices = distances.ravel(), indices.ravel()
+        reference_distances = self._get_reference_distances(main, indices, distances)
+        rescaled_distances = self._rescale_distances(distances, reference_distances)
+        return {
+            "index": indices,
+            "distance": distances,
+            "rescaled_distance": rescaled_distances,
+            "match_accepted": rescaled_distances <= max_dist,
+        }
+
+    def _get_reference_distances(self, main, indices, distances):
+        del main, indices, distances
```

Comment:
to make it clear (to readers and eventual static checkers & linters) that I intentionally don't use them (they are here for compatibility with other reference distances that need them). BTW this random pairs strategy doesn't need any of that info, it doesn't even require to vectorize the main table for `fit`, so if we ever make this one the only available strategy it would simplify the code a bit and have performance benefits